aptos-move/block-executor/src/executor.rs

```diff
@@ -1288,6 +1295,79 @@ where
                         "Sequential execution must materialize deltas"
                     );
 
+                    if resource_group_bcs_fallback {
+                        // Dynamic change set optimizations are enabled, and resource group serialization
+                        // previously failed in bcs serialization for preparing final transaction outputs.
+                        // TODO: remove this fallback when txn errors can be created from block executor.
+
+                        // The IDs are not exchanged but it doesn't change the types (Bytes) or size.
+                        let serialization_error = output
+                            .group_reads_needing_delayed_field_exchange()
+                            .iter()
+                            .any(|(group_key, _)| {
+                                fail_point!("fail-point-resource-group-serialization", |_| {
+                                    true
+                                });
+
+                                let finalized_group: BTreeMap<_, _> = unsync_map
+                                    .finalize_group(group_key)
+                                    .map(|(resource_tag, value_with_layout)| {
+                                        (
+                                            resource_tag,
+                                            value_with_layout
+                                                .extract_value_no_layout()
+                                                .extract_raw_bytes()
+                                                .expect("Deletions should already be applied"),
```

Comment:
Should we propagate this as an error?

Reply:
it's an easier / local invariant of unsync map